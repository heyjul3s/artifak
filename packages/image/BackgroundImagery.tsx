import * as React from 'react';
import { Base } from './typings';
import { getBackgroundImageStyles } from './utils/styles';
import { BackgroundImageryBase } from './BackgroundImageryBase';
import { ImgContext } from './context';

export function BackgroundImagery() {
  const {
    ariaLabel,
    ariaLabelledBy,
    ariaDescribedBy,
    children,
    fit,
    position,
    role,
    src,
    style,
  } = React.useContext<Base.BackgroundImagery>(ImgContext);

  const backgroundImageStyles = getBackgroundImageStyles({
    fit,
    position,
    style,
    src,
  });

  return (
    <BackgroundImageryBase
      aria-label={ariaLabel}
      aria-describedby={ariaDescribedBy}
      aria-labelledby={ariaLabelledBy}
      role={role}
      {...backgroundImageStyles}
    >
      {children}
    </BackgroundImageryBase>
  );
}
