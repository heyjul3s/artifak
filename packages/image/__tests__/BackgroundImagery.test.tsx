import React from 'react';
import { render } from '@testing-library/react';
import { BackgroundImagery } from '../BackgroundImagery';

describe('BackgroundImageryBase', () => {
  it('should render', () => {
    const { asFragment } = render(<BackgroundImagery />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
