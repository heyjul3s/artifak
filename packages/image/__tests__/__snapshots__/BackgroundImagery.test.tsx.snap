// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImageryBase should render 1`] = `
<DocumentFragment>
  <div
    aria-describedby=""
    aria-label=""
    aria-labelledby=""
    class="artifak__sc-6om8fa-0 xUprw"
    role=""
  />
</DocumentFragment>
`;
